dev/com.ibm.ws.jdbc/src/com/ibm/ws/jdbc/internal/JDBCDriverService.java

```diff
@@ -1053,6 +1055,28 @@ private void shutdownDerbyEmbedded() {
         else if (trace && tc.isEntryEnabled())
             Tr.exit(this, tc, "shutdownDerbyEmbedded", false);
     }
+    
+    private PropertyDescriptor[] getArrangedProperties(Class<?> dsClass) throws IntrospectionException {
+        PropertyDescriptor[] props = Introspector.getBeanInfo(dsClass).getPropertyDescriptors();
+        // Some JDBC drivers, such as PostgreSQL, implement setURL() in a way that calls other setXXX methods internally
+        // Make sure we arrange the URL property *last* so that setURL() gets called last and things like default 
+        // serverName or portNumber in metatype do not override a user-defined URL containing the same information
+        Arrays.sort(props, new Comparator<PropertyDescriptor>() {
+            @Override
+            public int compare(PropertyDescriptor p1, PropertyDescriptor p2) {
+                if (p1.getName().equalsIgnoreCase("url")) {
+                    if (p2.getName().equalsIgnoreCase("url"))
+                        return p1.getName().compareTo(p2.getName()); 
+                    else
+                        return 1;
+                }
+                if (p2.getName().equalsIgnoreCase("url"))
```

Comment:
If we are only doing this for PostgreSQL, do we really need the equalsIgnoreCase?  I imagine its more costly than normal equals.

Reply:
other JDBC drivers may benefit from this as well, so I kept it generic. I don't think the performance difference will be significant.